opentelekomcloud/services/ces/resource_opentelekomcloud_ces_alarmrule.go

```diff
@@ -387,55 +368,38 @@ func resourceAlarmRuleRead(ctx context.Context, d *schema.ResourceData, meta int
 	}
 	metricInfo := []map[string]interface{}{
 		{
-			"namespace":   r.Metric.Namespace,
-			"metric_name": r.Metric.MetricName,
+			"namespace":   alarm.Metric.Namespace,
+			"metric_name": alarm.Metric.MetricName,
 			"dimensions":  dimensionInfoList,
 		},
 	}
 
 	conditionInfo := []map[string]interface{}{
 		{
-			"period":              r.Condition.Period,
-			"filter":              r.Condition.Filter,
-			"comparison_operator": r.Condition.ComparisonOperator,
-			"value":               r.Condition.Value,
-			"unit":                r.Condition.Unit,
-			"count":               r.Condition.Count,
+			"period":              alarm.Condition.Period,
+			"filter":              alarm.Condition.Filter,
+			"comparison_operator": alarm.Condition.ComparisonOperator,
+			"value":               alarm.Condition.Value,
+			"unit":                alarm.Condition.Unit,
+			"count":               alarm.Condition.Count,
 		},
 	}
 
-	alarmActionsInfo := make([]map[string]interface{}, len(r.AlarmActions))
-	for i, alarmActionItem := range r.AlarmActions {
+	alarmActionsInfo := make([]map[string]interface{}, len(alarm.AlarmActions))
+	for i, alarmActionItem := range alarm.AlarmActions {
 		notificationList := make([]string, len(alarmActionItem.NotificationList))
-		for j, notification := range alarmActionItem.NotificationList {
-			notificationList[j] = notification
-		}
+		copy(notificationList, alarmActionItem.NotificationList)
 		alarmAction := map[string]interface{}{
 			"type":              alarmActionItem.Type,
 			"notification_list": notificationList,
 		}
 		alarmActionsInfo[i] = alarmAction
 	}
 
-	insufficientActionsInfo := make([]map[string]interface{}, len(r.InsufficientdataActions))
-	for i, insufficientActionItem := range r.InsufficientdataActions {
-		notificationList := make([]string, len(insufficientActionItem.NotificationList))
-		for j, notification := range insufficientActionItem.NotificationList {
-			notificationList[j] = notification
-		}
-		insufficientAction := map[string]interface{}{
-			"type":              insufficientActionItem.Type,
-			"notification_list": notificationList,
-		}
-		insufficientActionsInfo[i] = insufficientAction
-	}
-
-	okActionsInfo := make([]map[string]interface{}, len(r.OkActions))
-	for i, okActionItem := range r.OkActions {
+	okActionsInfo := make([]map[string]interface{}, len(alarm.OkActions))
+	for i, okActionItem := range alarm.OkActions {
```

Comment:
`i` is unused, should be replaced with `_`

Reply:
But it's used a bit later, otherwise it wouldn't compile

Reply:
My bad, haven't noticed